components/mediators/datamapper/org.wso2.carbon.mediator.datamapper.engine/src/main/java/org/wso2/carbon/mediator/datamapper/engine/core/mapper/XSLTMappingResource.java

```diff
@@ -0,0 +1,172 @@
+        if (notXSLTCompatible) {
+            this.content = null;
+        }
+    }
+
+    /**
+     * Create a input stream from the available content
+     *
+     * @return Input stream of the xslt stylesheet
+     */
+    InputStream getInputStream() {
+        return new ByteArrayInputStream(content.getBytes(StandardCharsets.UTF_8));
+    }
+
+    /**
+     * Create a input source from the available content
+     *
+     * @return Input source of the xslt stylesheet
+     */
+    private InputSource getInputSource() {
+        return new InputSource(new StringReader(content));
+    }
+
+    /**
+     * Creating a document to process the xslt stylesheet
+     *
+     * @return document of the xslt stylesheet
+     * @throws SAXException
+     * @throws IOException
+     * @throws ParserConfigurationException
+     */
+    private Document getDocument() throws SAXException, IOException,
+            ParserConfigurationException {
+        DocumentBuilderFactory documentBuilderFactory = DocumentBuilderFactory.newInstance();
+        DocumentBuilder documentBuilder = documentBuilderFactory.newDocumentBuilder();
+        return documentBuilder.parse(getInputSource());
+    }
+
+    /**
+     * Process configuration details included in the xslt stylesheet
+     *
+     * @param document xslt stylesheet as a document
+     * @return return whether the xslt transformation possible or not
+     */
+    private boolean processConfigurationDetails(Document document) {
+        boolean not_compatible = true;
+        Node rootNode = document.getElementsByTagName(PARAMETER_FILE_ROOT).item(0);
+        for (int j = 0; j < rootNode.getAttributes().getLength(); j++) {
+            Node propertyNode = rootNode.getAttributes().item(j);
+            switch (propertyNode.getNodeName()) {
+                case RUN_TIME_PROPERTIES:
+                    String runTimePropertyString = propertyNode.getNodeValue();
+                    if (!EMPTY_STRING.equals(runTimePropertyString)) {
+                        String[] properties = runTimePropertyString.split(PROPERTY_SEPERATOR);
+                        int currentIndex = 0;
+                        while (currentIndex < properties.length) {
+                            runTimeProperties.put(properties[currentIndex],
+                                    properties[currentIndex + 1]);
+                            currentIndex += 2;
+                        }
+                    }
+                    break;
+                case NOT_XSLT_COMPATIBLE:
+                    if (propertyNode.getNodeValue().equals(XSLT_COMPATIBLE_DEFAULT)) {
+                        not_compatible = false;
+                    } else {
+                        return not_compatible;
+                    }
+                    break;
+                case FIRST_ELEMENT_OF_THE_INPUT:
+                    this.name = propertyNode.getNodeValue();
+                    break;
+            }
```

Comment:
What'd happen if the node name does not belong to the above three cases? Don't we need a default case?

Reply:
There are many others nodes (properties such as XSLT version). But I only need to process three properties. Therfore, i am not using a default case.